### DISCUSSION TITLE
Use of grouped data, regressors and auxiliary series.

### DISCUSSION THREAD

**Discussion by Marcovnr** [winedarksea/AutoTS#148] [Q&A]
Hello!

I am currently using AutoTS for timeseries problems and I am loving it. First of all I wanted to say congratulations for the awesome library you are building.

I´ve come here because I am facing a specific problem and I am not sure of how to approach it.

I have a dataframe that contains some dimensions that should be considered while training. What I mean by dimensions is almost a groupby kind of thing, for example for a sales record we may have dimensions of products and stores (you may have a series for product 1 in store 1, product 2 in store 1, produtc1 in store 2...).

If that was it, it would be okay to treat all as simply different series, however I have a few regressors (data that I am certain about the future) and few aux variables, and some of these are passed in a dimension level (for example a variable indicating weather in store 1, and another one indicating weather in store 2).

With that in mind, I am not sure of how to proceed with this. Unfortunately, due to some external reasons, I have to include all of these in one single run. I came up with these alternatives:

1. Ungroup the aux variables to represent the various dimensions of the problem, creating different series for each one of these new columns.
2. Use the dimensions as regressors, instead of series_id.

Besides that, I would also make use that I am already writing this post, and ask if the weights parameter in the .fit() method can be used with long data.
I would really use some help. I really appreciate you guys for any given.

**Comment by winedarksea** [comment]
@Marcovnr  Thanks for all the compliments, always good to hear!

You can use as many regressors as you want in the AutoTS class, so just 'throw everything all in at once as regressors' is a perfectly fine approach. Generally the models can learn feature weights to essentially ignore regressors that aren't relevant to a particular series. 
Pass in all the time series, one for each product and store, as your data, then as regressors pass anything you think might be useful, weather for all stores and dimension data, etc. It usually works well enough.

You might be interested in the new Cassandra model. While unfortunately not integrated into the higher level parameter tuning, it has a a `regressor_per_series` dictionary and also `categorical_groups` option to .fit which look to do exactly what you need. There's an example at the end of the autots.models.cassandra.py file.

Weights can be used with long data identically to wide data, as they are not used until after long day is internally transformed into wide data.

this was a quick answer, I hope it isn't too confusing

**Reply by Marcovnr** [reply]
Hi Catlin! 

Thanks so much for the answer, I believe I understood everything correctly! It was not confusing at all! 

Thanks for the quick response! If anything comes up again regarding this discussion, I will comment it here!
